src/System.Text.Json/src/System/Text/Json/Serialization/Converters/JsonPropertyInfoChar.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace System.Text.Json.Serialization.Converters
+{
+    internal class JsonPropertyInfoChar : JsonPropertyInfo<char>, IJsonValueConverter<char>
+    {
+        public JsonPropertyInfoChar(Type classType, Type propertyType, PropertyInfo propertyInfo, JsonSerializerOptions options) :
+            base(classType, propertyType, propertyInfo, options)
+        { }
+
+        public bool TryRead(Type valueType, ref Utf8JsonReader reader, out char value)
+        {
+            if (reader.TokenType != JsonTokenType.String)
+            {
+                value = default;
+                return false;
+            }
+
+            value = reader.GetString()[0];
+            return true;
+        }
+
+        public void Write(char value, ref Utf8JsonWriter writer)
+        {
+#if BUILDING_INBOX_LIBRARY
+            Span<char> temp = MemoryMarshal.CreateSpan<char>(ref value, 1);
+            writer.WriteStringValue(temp);
+#else
+            writer.WriteStringValue(value.ToString());
```

Comment:
You shouldn't need to ToString() in these cases: you can instead just stackalloc and copy to that, then write that span, e.g.
```C#
Span<char> span = stackalloc char[1] { value };
writer.WriteStringValue(span);
```

Reply:
It look like the compiler now detects stackalloc:
```
Error	CS8350	This combination of arguments to 'Utf8JsonWriter.WriteStringValue(ReadOnlySpan<char>, bool)' is disallowed because it may expose variables referenced by parameter 'value' outside of their declaration scope```

Reply:
Ugh, because the writer is a ref struct and is being passed around by ref.

Reply:
It just needs to be unsafe. @JeremyKuhne ran into this as well.

Reply:
Specifically David means:
```C#
Span<char> span;
unsafe
{
    char* tmp = stackalloc char[size];
    span = new Span<char>(tmp, size);
}
```
but if you do that, you need to be very careful that this span doesn't get returned accidentally.

Reply:
@jaredpar not news, but the non-returnable thing is a constant pain :)